src/codec.rs

```diff
@@ -665,7 +667,7 @@ macro_rules! impl_array {
 }
 
 impl_array!(
-	1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16,
+	0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16,
```

Comment:
it should not require `T: Encode + Decode` though for empty arrays, right?

Reply:
While that is conceptually right, why would you want to write such code? Than you directly could use `()`.

Reply:
Good point, I don't know what @Robbepop use-case is.

Reply:
Before we merge this: I was able to resolve the problem without having support for SCALE encoded empty arrays. If you still want to merge this, go ahead.

Reply:
Fine to merge anyway :)

Reply:
Yes, it is a nice generalization imo.